nova_vm/src/ecmascript/builtins/indexed_collections/typed_array_objects/typed_array_intrinsic_object.rs

```diff
@@ -3307,13 +3308,111 @@ impl TypedArrayPrototype {
+            .unbind()?
+            .bind(gc.nogc());
+        let o = match ta_record.object {
+            TypedArray::Int8Array(_) => with_typed_array::<i8>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            TypedArray::Uint8Array(_) => with_typed_array::<u8>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            TypedArray::Uint8ClampedArray(_) => with_typed_array::<U8Clamped>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            TypedArray::Int16Array(_) => with_typed_array::<i16>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            TypedArray::Uint16Array(_) => with_typed_array::<u16>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            TypedArray::Int32Array(_) => with_typed_array::<i32>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            TypedArray::Uint32Array(_) => with_typed_array::<u32>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            TypedArray::BigInt64Array(_) => with_typed_array::<i64>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            TypedArray::BigUint64Array(_) => with_typed_array::<u64>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            #[cfg(feature = "proposal-float16array")]
+            TypedArray::Float16Array(_) => with_typed_array::<f16>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            TypedArray::Float32Array(_) => with_typed_array::<f32>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+            TypedArray::Float64Array(_) => with_typed_array::<f64>(
+                agent,
+                ta_record.unbind(),
+                index.unbind(),
+                value.unbind(),
+                gc,
+            ),
+        };
```

Comment:
This can be simplified to (#689):
```suggestion
        let o = with_typed_array_viewable!(
          ta_record.object,
          with_typed_array::<T>(
            agent,
            ta_record.unbind(),
            index.unbind(),
            value.unbind(),
            gc,
          )
        );
```

Reply:
You’re awesome, thank you so much!!!